File: accsaber-business/src/main/java/de/ixsen/accsaber/business/exceptions/AccsaberOperationException.java
package de.ixsen.accsaber.business.exceptions;

public class AccsaberOperationException extends RuntimeException {

    private final String errorCode;

    public AccsaberOperationException(ExceptionType exceptionType, String message) {
        super(message);
        this.errorCode = exceptionType.getErrorCode();
    }

    public AccsaberOperationException(ExceptionType exceptionType, String message, Throwable cause) {
        super(message, cause);
        this.errorCode = exceptionType.getErrorCode();
    }

    public String getErrorCode() {
        return errorCode;
    }
}

File: accsaber-business/src/main/kotlin/de/ixsen/accsaber/business/exceptions/AccsaberOperationException.kt
package de.ixsen.accsaber.business.exceptions

class AccsaberOperationException : RuntimeException {
    val errorCode: String

    constructor(exceptionType: ExceptionType, message: String) : super(message) {
        errorCode = exceptionType.errorCode
    }

    constructor(exceptionType: ExceptionType, message: String, cause: Throwable) : super(message, cause) {
        errorCode = exceptionType.errorCode
    }
}